src/services/stats/stats-router.test.ts

```diff
@@ -0,0 +1,447 @@
+import { beforeEach, describe, expect, it } from "@jest/globals";
+import { StatusCodes } from "http-status-codes";
+import { Database } from "../../database";
+import { Role } from "../auth/auth-models";
+import { getAsStaff, get } from "../../../testing/testingTools";
+import { getCurrentDay } from "../checkin/checkin-utils";
+
+describe("GET /stats/check-in", () => {
+    beforeEach(async () => {
+        await Database.ATTENDEE.deleteMany({});
+    });
+
+    it("should return correct count for checked-in attendees", async () => {
+        await Database.ATTENDEE.create([
+            {
+                userId: "a1",
+                name: "Ritam",
+                email: "ritam@test.com",
+                hasCheckedIn: true,
+            },
+            {
+                userId: "a2",
+                name: "Nathan",
+                email: "nathan@test.com",
+                hasCheckedIn: true,
+            },
+            {
+                userId: "a3",
+                name: "Timothy",
+                email: "timothy@test.com",
+                hasCheckedIn: false,
+            },
+        ]);
```

Comment:
Don't put the raw objects in the tests, extract into global constants at the top of the file. Also, the best practice is to create all 3 in a beforeEach, then delete the ones you don't want in the specific tests. The goal is minimizing duplication.